pkg/tarmak/cluster/cluster.go

```diff
@@ -297,6 +305,33 @@ func (c *Cluster) validateAPIServer() (result error) {
 	return result
 }
 
+func (c *Cluster) validatePrometheusMode() error {
+	var result error
+
+	allowedModes := []string{
+		clusterv1alpha1.PrometheusModeFull,
+		clusterv1alpha1.PrometheusModeExternalScrapeTargetsOnly,
+		clusterv1alpha1.PrometheusModeExternalExportersOnly,
+	}
```

Comment:
Drive by code review...

`sets.NewString` might be useful here.
*  https://github.com/kubernetes/apimachinery/blob/master/pkg/util/sets/string.go

```
allowedModes := sets.NewString(
    clusterv1alpha1.PrometheusModeFull, 
    clusterv1alpha1.PrometheusModeExternalScrapeTargetsOnly,
    clusterv1alpha1.PrometheusModeExternalExportersOnly,
)

if ! allowedModes.Has(modeString) {
...
   fmt.Errorf("allowed modes: %s", allowedModes.List()
...
}
```

Reply:
Thanks - this is much nicer. Should be fixed in c0c771d647a9ff967b10637fe18c4f4a342aaa4e